doozerlib/distgit.py

```diff
@@ -941,11 +904,23 @@ def _build_container(
             "--nowait",
         )
 
-        if odcs:
-            if odcs == 'signed':
-                odcs = 'release'  # convenience option for those used to the old types
-            cmd_list.append('--signing-intent')
-            cmd_list.append(odcs)
+        # Determine if ODCS is enabled by looking at container.yaml.
+        odcs_enabled = False
+        osbs_image_config_path = os.path.join(self.distgit_dir, "container.yaml")
+        if os.path.isfile(osbs_image_config_path):
+            with open(osbs_image_config_path, "r") as f:
+                image_config = yaml.safe_load(f)
+            odcs_enabled = "compose" in image_config
+
+        if odcs_enabled:
+            self.logger.info("About to build image in ODCS mode.")
+            if repo_type == 'signed':
+                signing_intent = 'release'
```

Comment:
there could be a use case for allowing beta here... deal with it if it comes up i guess.